saleor/checkout/complete_checkout.py

```diff
@@ -1231,6 +1231,7 @@ def _create_order_from_checkout(
 
     # voucher
     voucher = checkout_info.voucher
+    voucher_code = checkout_info.checkout.voucher_code
```

Comment:
```suggestion
    voucher_code = (
        checkout_info.voucher_code.code if checkout_info.voucher_code else None
    )
```
Checkouts are processed with prefetched data to prevent race conditions. Let's use the code from `checkout_info.voucher_code` instance.

Reply:
Thanks @zedzior